Closes #358

apragsdale opened the pull request:
~~Check if "Infinity" is given as a start time, which is converted to `math.inf` in `Builder.add_deme()`. Maybe there is a more appropriate place to handle this?~~ What do you think, @grahamgower 

~~Edit: uses `load_dump._unstringify_infinities()` in `Builder.resolve()`.~~

Edit 2: simply adjust a deme or migration start time in `add_deme()` and `add_migration()`. Using `_unstringify_infinities()` causes some complications where `fromdict()` in `Builder.resolve()` is not able to catch type errors.

grahamgower commented:
What you have here looks right, thanks @apragsdale. In #358, I mentioned that the `defaults` fields are not checked for string infinity value---I guess just `defaults["deme"]["start_time"]`? Maybe there are other fields that could have infinities that I didn't think of when writing _unstringify_infinities()?

Oh, and when you squash, would you be able to change the commit message to indicate what the change is doing (rather than just referring to the issue number)?

apragsdale commented:
Closes #358.

Ok, updated here. Thanks for the suggestions. I'll wait a bit and then merge unless you think there is something else that is needed here.

> What you have here looks right, thanks @apragsdale. In #358, I mentioned that the `defaults` fields are not checked for string infinity value---I guess just `defaults["deme"]["start_time"]`? Maybe there are other fields that could have infinities that I didn't think of when writing _unstringify_infinities()?

Also added a bit in `_ustringify_infinities()` to check default start times in demes and migrations.

> Oh, and when you squash, would you be able to change the commit message to indicate what the change is doing (rather than just referring to the issue number)?

Good point. Commit message is changed.
